### ISSUE TITLE
Add Docker Installation/Upgrading To Docs

### ISSUE THREAD

**Issue by 9mido** [python-poetry/poetry#2822]
Stackoverflow and some other sites have methods for Poetry to be installed with Docker. But they are not all noob friendly and might not be the best ways to do things. 

Please add to the docs the recommended way to install poetry with docker and how to upgrade poetry with docker.

**Comment by tetienne** [comment]
To understand clearly, you want to run poetry through a docker container, or build an image with Poetry inside?

**Comment by 9mido** [comment]
@tetienne Good point. I now see #1879 which has best practices and just realized this may be a duplicate of that issue. 

There are quite a few use cases to account for like you mentioned. That includes regular container run commands, Dockerfile and YML configurations. Accounting for the most common use cases in the Poetry documentation would be ideal.

**Comment by max-pfeiffer** [comment]
Have a look at these two Docker projects, they provide base images for dockerizing your web application with Poetry:

- https://github.com/max-pfeiffer/uvicorn-poetry
- https://github.com/max-pfeiffer/uvicorn-gunicorn-poetry
